### ISSUE TITLE
[Bug] Repeatedly upscaling outputs pixels when X/Y/Z script is used

### ISSUE THREAD

**Issue by wklchris** [pkuliyi2015/multidiffusion-upscaler-for-automatic1111#36]
I found this bug when I tried to upscale a 512x768 image by x1.5 and compared different parameters. To do this, I enabled the built-in `X/Y/Z plot` script in img2img tab of WebUI, then set X to 2 `CFG Scale` values and Y to 2 `Denoising` values. 

#### Expected results
I expected to get 2x2=4 images with their pixels all in 768x1152, which is x1.5 upscaled over the input. 

#### Actual results
However, I get images pixels upscaled over and over again. I checked the img2img outputs folder and find things like:

- 1st image in 768x1152 (correct, x1.5 over the input image)
- 2nd image in 1152x1728 (incorrect, x1.5 over the 1st image, which is actually x2.25 over the input)
- 3rd image in 1728x2592
- …… (I interrupted the process here because it takes so long to finish)

#### Test Environment

The above bug is confirmed with the latest Multidiffusion extension commit (23ccdc748eba614f8f0f2fd4a632db31f82cc86d) and WebUI commit ([a9fed7c](https://github.com/AUTOMATIC1111/stable-diffusion-webui/commit/a9fed7c364061ae6efb37f797b6b522cb3cf7aa2)).

**Comment by pkuliyi2015** [comment]
Noticed with thanks. To avoid this problem I may need to do some magics with XYZ plot... I don't have much understand on this, will look into the controlnet xyz plot scripts.

**Comment by altoiddealer** [comment]
I just want to chime in that this effect only happens when "Keep input image size" is enabled.

When the setting is disabled, the X/Y/Z plot runs as expected.

This is also unrelated to Controlnet.


So - maybe you can find a fix for this while "Keep input image size" is enabled.

**Comment by Kahsolt** [comment]
Thanks for the report, checked and found the reason. 
The X/Y/Z plot calls tilediff multiple times, so tilediff does multiple upscaling on the given ref-img.

To trace this thing is not elegant, switching off the "Keep input image size" option is already the best answer ;)

**Comment by altoiddealer** [comment]
Oh wow, long followup on this one lol.  I’m not so sure that “keep input size” actually resolves it… when I wrote the comment, that seemed to work, but later on I tried against and it didn’t work.  I can’t test right now… it could be that I forgot to select an upscaler the first time or something.

one thing is for sure though, when I use your extension it usually takes me a lot of guess and check to find the ideal settings for my project.  x/y/z is pretty important for this extension
